### PR TITLE
Honor XKB_CONFIG_ROOT environment variable, don't hardcode

### PR DESCRIPTION
**Pull request by worldofpeace**
PR on behalf of nixpkgs, this has been a problem for us in NixOS
for gnome-desktop [0] and other projects. See aforementioned issue
for more details.

[0]: https://gitlab.gnome.org/GNOME/gnome-desktop/issues/130

### CODE REVIEW

File: src/Layout/Handler.vala [hunk 1]
```diff
@@ -28,10 +30,23 @@ public class Pantheon.Keyboard.LayoutPage.LayoutHandler : GLib.Object {
         languages = new HashTable<string, string> (str_hash, str_equal);
     }
 
+    public string? get_xml_rules_file_path () {
+        string base_path = GLib.Environment.get_variable ("XKB_CONFIG_ROOT");
```
Comment (tintou) [inline src/Layout/Handler.vala]:
```suggestion
        unowned string? base_path = GLib.Environment.get_variable ("XKB_CONFIG_ROOT");
```

File: src/Layout/Handler.vala [hunk 2]
```diff
@@ -28,10 +30,23 @@ public class Pantheon.Keyboard.LayoutPage.LayoutHandler : GLib.Object {
...
+
+        if (base_path == null) {
+            base_path = Constants.XKB_BASE;
+        }
+
+        string rules_file = Path.build_filename (base_path, "rules", XKB_RULES_FILE);
```
Comment (tintou) [inline src/Layout/Handler.vala]:
```suggestion
        return Path.build_filename (base_path, "rules", XKB_RULES_FILE);
```

File: src/Layout/Handler.vala [hunk 3]
```diff
@@ -28,10 +30,23 @@ public class Pantheon.Keyboard.LayoutPage.LayoutHandler : GLib.Object {
         languages = new HashTable<string, string> (str_hash, str_equal);
     }
 
+    public string? get_xml_rules_file_path () {
+        string base_path = GLib.Environment.get_variable ("XKB_CONFIG_ROOT");
+
```
Comment (tintou) [inline src/Layout/Handler.vala]: You don't need a newline here :)

File: src/Layout/Handler.vala [hunk 4]
```diff
@@ -28,10 +30,23 @@ public class Pantheon.Keyboard.LayoutPage.LayoutHandler : GLib.Object {
         languages = new HashTable<string, string> (str_hash, str_equal);
     }
 
+    public string? get_xml_rules_file_path () {
```
Comment (tintou) [inline src/Layout/Handler.vala]:
```suggestion
    public string get_xml_rules_file_path () {
```

File: src/Layout/Handler.vala [hunk 5]
```diff
@@ -28,10 +30,23 @@ public class Pantheon.Keyboard.LayoutPage.LayoutHandler : GLib.Object {
...
+        }
+
+        string rules_file = Path.build_filename (base_path, "rules", XKB_RULES_FILE);
+
+        return rules_file;
+    }
+
     private void parse_layouts () {
-        Xml.Doc* doc = Xml.Parser.parse_file ("/usr/share/X11/xkb/rules/evdev.xml");
+        string file_path = get_xml_rules_file_path ();
```
Comment (jeremypw) [inline src/Layout/Handler.vala]: I don't think you need the string copy 'file_path'?  It appears to be used only once.
Suggest:
`Xml.Doc* doc = Xml.Parser.parse_file (get_xml_rules_file_path ());`

**Comment by worldofpeace** [comment]
Thanks for the review, this code is exact in another PR for wingpanel-indicator-keyboard so I will apply the same changes there.

**Comment by worldofpeace** [comment]
Requested changes applied in the fixup commit.

**Pull request merged**